README.md

```diff
@@ -1 +1,168 @@
-# redirects
\ No newline at end of file
+# Url Redirects with GitHub
+
+_Counterpart to the [redirects-website repo](../../../redirects-website)._
+
+✏️ [How to edit](#how-to-edit)  
+💡 [Motivation](#motivation)  
+✔️ [Comparison](#comparison)  
+🧠 [How it works](#how-it-works)  
+⚙️ [Setup](#setup)
+
+## How to edit
+
+1. Add/change/remove redirect entries in one or more [`.yaml` files in the top folder](../../blob/main/redirects.yaml).
+2. Commit the changes to the `main` branch.
+3. Changes should take effect automatically within a minute or so.
+   Verify that no errors occurred in the automatic process here: [![Encode and deploy](../../actions/workflows/deploy.yaml/badge.svg)](../../actions/workflows/deploy.yaml)
+
+You can do this [directly on github.com](../../edit/main/redirects.yaml) (tip: press <kbd>.</kbd> right now), or locally with git.
+
+## Motivation
+
+You've likely heard of a service called [Bitly](https://bitly.com/).
+It allows you to convert a long link like `some-website.com/a-long-url?search=a-bunch-of-characters` into a shorter one like `bit.ly/98K8eH`.
+When someone visits the shorter link, Bitly automatically "redirects" them to the longer one.
+You can think of it like a shortcut.
+
+Bitly also offers several other features that you probably want:
+
+- You can customize the text after the `/`, giving you a url that a human could actually remember and type in manually, like `bit.ly/MyCoolLink` (they call these "back-halves").
+- You can set up a custom domain to brand your links the way you want, giving you an even nicer and easier-to-use link, like `my-website.com/MyCoolLink`.
+- You can see how many and what kinds of people have used the link over time (i.e. analytics).
+
+This all sounds great, so what's the problem?
+Well namely: _Bitly [hides or limits a lot of this functionality behind a paywall](https://bitly.com/pages/pricing)_, and their free plan has become more and more limited over time.
+This may not be a problem for you, especially if you value the convenience of having a simple service that handles everything automatically.
+And there are certainly [other services that compete with Bitly](https://zapier.com/blog/best-url-shorteners/) which may offer you what you want for free (for now).
+
+But with just a little bit of setup, we can accomplish all of this in a much better way.
+Well, at least much better-suited to the _target audience of this approach_: **People/organizations who use GitHub and Git**.
+
+## Comparison
+
+Here's how this approach compares to Bitly and similar services.
+
+**The benefits**:
+
+- Free(er)!
+  You only need to pay for a custom domain name, if you want.
+- Not subject to the pricing whims of large companies.
```

Comment:
I think GitHub Pages pricing has been the same for as long as I've known about it, whereas these url services seem to change their tiers all the time.

I'll change it to "pricing whims of the url shortening services".

Reply:
GitHub Pages has been free since the creation of GitHub in 2008:
https://en.wikipedia.org/wiki/Timeline_of_GitHub

So I feel confident not calling it out specifically, and sticking with "pricing and features _should_ remain the same".

Reply:
@vincerubinetti - thanks, I noticed the update here; it feels more clear now, appreciate it.